napari/_vispy/overlays/scale_bar.py

```diff
@@ -17,6 +17,9 @@ def __init__(self, **kwargs) -> None:
         self._scale = 1
         self._unit = None
 
+        # TODO create neater way to access background_color_override
+        self.canvas = kwargs['canvas']
+        kwargs.pop("canvas")
```

Comment:
```suggestion
        self.canvas = kwargs.pop("canvas")
```

Or better add `canvas=None` to the init args and then `self.canvas = canvas`

But this will be soon different in #4894.

Reply:
Yes I was aware of this PR as we discussed it and did only touch overlays to the minimum possible for that reason.

Reply:
Given the 24h clock for merging the mentioned PR, I prefer merging mentioned PR from main.